test/functional/p2p_invalid_messages.py

```diff
@@ -125,6 +128,22 @@ def test_msgtype(self):
             conn.sync_with_ping(timeout=1)
         self.nodes[0].disconnect_p2ps()
 
+    def test_inv_msg_with_signed_integer_type(self):
+        self.log.info("Test inv message with MSG_TX type as signed int raises out of range error")
+        inv_with_signed_type = msg_inv([CInv(t=-MSG_TX, h=1)])
+        try:
+            self.nodes[0].add_p2p_connection(P2PInterface()).build_message(inv_with_signed_type)
+        except struct.error as e:
```

Comment:
imo could drop the test framework self-test

Reply:
Done, but I think both of these tests only tested the framework change anyway.